### PR TITLE
(Updated) Add ja-JP locale to 3.9 release

### PR DESCRIPTION
**Pull request by Cai0407**
I have created new PR to add ja-JP locale to 3.9 release.
(Previous PR #200 is conflicted with current master)

### CODE REVIEW

File: extension/chrome.manifest [hunk 1]
```diff
@@ -2,6 +2,7 @@ content      nightly                                                  chrome/con
 skin         nightly          classic/1.0                             chrome/skin/
 
 locale       nightly          en-US                                   chrome/locale/en-US/
+locale       nightly          ja-JP                                   chrome/locale/ja-JP/
```
Comment (whimboo) [inline extension/chrome.manifest]: For Firefox we have `ja`and `ja-JP-mac`. Shouldn't it be also reflected this way here in the extension? The localized files we could place under `chrome/locale/ja`.

**Comment by whimboo** [comment]
Regarding proof-reading of the localization I sent an email to Masahiko, who is one of the maintainers of the ja locale and browser/toolkit peer.

**Comment by marsf** [comment]
OK, I'll check them.

File: extension/chrome.manifest (hunk 1 above)
Comment (marsf) [inline extension/chrome.manifest]: ```ja-JP``` is ok for both ja and ja-JP-mac. But ```ja``` is preferred.

File: extension/chrome/locale/ja-JP/customize.dtd [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+<!-- LOCALIZATION NOTE: The same English string as the following is also defined
+                        as a different entity for use in the menu,
+                        see ./nightly.dtd -->
+<!ENTITY nightly.customize.title              "タイトルバーのカスタマイズ">
+
+<!ENTITY nightly.paneTitlebar.title           "タイトルバー">
+<!ENTITY nightly.paneTitlebar.description     "タイトルバーのカスタマイズ">
+
+<!ENTITY nightly.customtitle.label            "カスタマイズしたタイトルバーを使用する">
```
Comment (marsf) [inline extension/chrome/locale/ja-JP/customize.dtd]: -> カスタマイズしたタイトルを使用する

File: extension/chrome/locale/ja-JP/customize.dtd [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+<!ENTITY nightly.customtemplate.label         "書式">
```
Comment (marsf) [inline extension/chrome/locale/ja-JP/customize.dtd]: -> タイトルのテンプレート:

File: extension/chrome/locale/ja-JP/nightly.properties [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+extensions.{8620c15f-30dc-4dba-a131-7c5d20cf4a29}.description=開発版のテスト利用者のための有用なツールを提供します
+nightly.notextbox.message=この機能を使用する前にテキストボックスを選択する必要があります
+nightly.notextbox.clipboardInstead.message=あるいは、クリップボードにコピーすることもできます
+nightly.copyButton.message=クリップボードにコピー
+nightly.noextensions.message=アドオンが見つかりません
+nightly.restart.message=アドオンの互換性確認が変更されました。再起動してください
```
Comment (marsf) [inline extension/chrome/locale/ja-JP/nightly.properties]: -> アドオンの互換性確認が変更されました。再起動して変更を適用してください

File: extension/chrome/locale/ja-JP/screenshot.dtd [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+
+<!ENTITY screenshot.window.title        "スクリーンショットの撮影">
+
+<!ENTITY screenshot.save.label          "名前を付けて保存...">
```
Comment (marsf) [inline extension/chrome/locale/ja-JP/screenshot.dtd]: For **ja-JP-mac**: 別名で保存...

File: extension/chrome/locale/ja-JP/screenshot.dtd [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+<!ENTITY screenshot.submit.label        "ImageShack に投稿...">
+<!ENTITY screenshot.copy.label          "クリップボードにコピー">
+<!ENTITY screenshot.redraw.label        "再描画">
+<!ENTITY screenshot.zoom.label          "ズーム:">
+<!ENTITY screenshot.window.label        "ウインドウ:">
```
Comment (marsf) [inline extension/chrome/locale/ja-JP/screenshot.dtd]: For **ja**: ウィンドウ
and **ja-JP-mac**: ウインドウ

File: extension/chrome/locale/ja-JP/variables.properties [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+variable.Name.description=アプリケーションの名称
+variable.Version.description=アプリケーションのバージョン
+variable.AppBuildID.description=アプリケーションのビルド ID
+variable.Changeset.description=ビルド元のチェンジセット
+variable.PlatformBuildID.description=XUL プラットフォームのビルド ID
+variable.PlatformVersion.description=XUL プラットフォームのバージョン
+variable.GeckoBuildID.description=Gecko のビルド ID
+variable.GeckoVersion.description=Gecko のバージョン
+variable.BrandName.description=アプリケーションのブランド名
+variable.UserAgent.description=UA 文字列
```
Comment (marsf) [inline extension/chrome/locale/ja-JP/variables.properties]: -> ユーザーエージェント文字列

File: extension/chrome/locale/ja-JP/variables.properties [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+variable.Locale.description=現在の言語
+variable.OS.description=コンパイルの対象 OS
+variable.Processor.description=コンパイルの対象プロセッサ
+variable.Compiler.description=コンパイラ
+variable.DefaultTitle.description=現在のアプリケーションのタイトル
+variable.TabTitle.description=現在のタブのタイトル
+variable.Profile.description=現在のプロファイル
+variable.Toolkit.description=グラフィックツールキット
+variable.TabsCount.description=表示可能なタブの数
```
Comment (marsf) [inline extension/chrome/locale/ja-JP/variables.properties]: -> 表示されたタブの数

File: extension/install.rdf [hunk 9]
```diff
@@ -41,6 +41,11 @@
       <em:locale>pl</em:locale>
       <em:description>Użyteczne narzędzia dla testerów wersji Nightly.</em:description>
     </em:localized>
+    <em:translator>Kosuke Kaizuka (ja-JP)</em:translator>
+    <em:localized>
+      <em:locale>ja-JP</em:locale>
```
Comment (marsf) [inline extension/install.rdf]: ```ja``` is preferred.

**Comment by marsf** [comment]
There is some OS specific terms are used. It is good to make and separate to 2 locales (```ja``` and ```ja-JP-mac```).

**Comment by whimboo** [comment]
Thank you @marsf for checking the files. @Cai0407, do you think that you can take care of this soon? If it would take some days for you, I would like to move the milestone of this locale to the 3.10 release of NTT.

**Comment by Cai0407** [comment]
@marsf, thank you for reviewing.

I will also add line for `ja-JP-mac` in extension/chrome.manifest.
> locale       nightly          ja                                      chrome/locale/ja/
locale       nightly          ja-JP-mac                               chrome/locale/ja-JP-mac/

Is it good to make another `<em:localized>` section for `ja-JP-mac` in extension/install.rdf?
>     <em:translator>Kosuke Kaizuka (ja, ja-JP-mac)</em:translator>
    <em:localized>
      <em:locale>ja</em:locale>
      <em:description>開発版のテスト利用者のための有用なツールを提供します</em:description>
    </em:localized>
    <em:localized>
      <em:locale>ja-JP-mac</em:locale>
      <em:description>開発版のテスト利用者のための有用なツールを提供します</em:description>
    </em:localized>

@whimboo, I have already applied Masahiko's suggestions in my local. I'm ready to pushing it to remote after chacking 2 points above.

**Comment by whimboo** [comment]
I think that each locale needs an entry in the manifest, so yes please do so. Thanks.

**Comment by Cai0407** [comment]
Updated.

**Comment by marsf** [comment]
Entries in extension/chrome.manifest are required.

The ```<em:localized>``` section for ```ja-JP-mac``` is not needed unless it has any differences, because it will fall back to ```ja```  and default one.

**Comment by Cai0407** [comment]
Removed `<em:localized>` section for `ja-JP-mac` in install.rdf.

**Comment by whimboo** [comment]
@marsf, are those changes to the new dtd files are looking good to you? If that is the case we can get this PR merged. Thanks.

**Comment by marsf** [comment]
@wimboo, It looks good on Nightly, win10. I cannot test on mac, but it should be ok.

**Comment by whimboo** [comment]
This sounds great. Thanks for re-testing the changes! I will go ahead and merge this PR now. @Cai0407 thanks a lot for the ja addition.

**Pull request merged**